server/src/models/setlist.model.ts

```diff
@@ -0,0 +1,30 @@
+import { Schema, model, models, Types } from 'mongoose';
+
+interface ISetlist {
+  name: string;
+  date: string;
+  createdBy: Types.ObjectId;
+  songs: Types.Array<Types.ObjectId>;
+  lastUpdatedBy: Types.ObjectId;
+  publicLink: string;
+  groupIds: Types.Array<Types.ObjectId>;
+}
+
+const setlistSchema = new Schema<ISetlist>(
+  {
+    name: { type: String, required: true },
+    date: { type: String, required: true },
+    createdBy: { type: Schema.Types.ObjectId, ref: 'User' },
+    songs: [{ type: Types.ObjectId, ref: 'Song' }],
+    lastUpdatedBy: { type: Schema.Types.ObjectId, ref: 'User' },
+    publicLink: { type: String, required: true, unique: true },
```

Comment:
how are we going to produce this public link ya? better on frontend or backend?

Reply:
I think currently the link is smth like /setlist/:id. Do we wanna make it customizeable? Like /setlist/name?

Not sure which one is a better procedure in fe / be :"). Gut feeling says be tho lol cuz i rmb albert told me fe shodnt be doing any routing / db filter logic.

Reply:
i think customizeable can be for future

but yes, it should be backend then